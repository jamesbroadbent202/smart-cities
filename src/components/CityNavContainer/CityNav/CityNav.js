import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import style from './CityNav.scss';
import { NO_CITY } from '../../../constants';

const classnames = require('classnames/bind').bind(style);

const CityNav = (props) => {
  const className = classnames({
    [style.citySelected]: props.cityId !== NO_CITY,
  });

  return (
    <div className={className}>
      <NavLink
        className={classnames(style.link, style.allCitiesLink)}
        activeClassName={style.link__active}
        to={`/${NO_CITY}/${props.categoryId}`}
      >
        All Cities

        <span className={style.cityCount}>
          {props.cities.length}
        </span>
      </NavLink>

      {props.cities.map(city => (
        <NavLink
          key={city.id}
          className={classnames(style.link, style.cityLink)}
          activeClassName={style.link__active}
          to={`/${city.id}/${props.categoryId}`}
        >{city.name}</NavLink>
      ))}
    </div>
  );
};

CityNav.propTypes = {
  categoryId: PropTypes.string.isRequired,
  cities: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
  })).isRequired,
  cityId: PropTypes.string.isRequired,
};

export default CityNav;
