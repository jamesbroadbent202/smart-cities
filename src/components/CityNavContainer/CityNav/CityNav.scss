@import '~style/variables';

.link {
  @include type-nav-vertical;

  display: block;
  transition: $animation-duration;
  color: $color-white;

  .citySelected & {
    color: $color-primary--060;
  }

  &:hover {
    background: $color-primary--800;
    color: $color-white;
  }

  &.link__active {
    background: $color-primary--900;
    color: $color-white;
  }
}

.allCitiesLink {
  @include type-nav-vertical-large;

  display: flex;
  align-items: center;
  height: $top-nav-height-mobile;
  margin-bottom: 6px;
  padding: 0 16px;
  background: $color-primary--800;

  .citySelected & {
    color: $color-primary--080;
  }

  @include for-desktop-up {
    height: $top-nav-height-desktop;
  }
}

.cityLink {
  padding: 0 10px 0 32px;
  line-height: 40px; // height is set because when selected this turns bold

  &.link__active {
    font-weight: 700;
  }
}

.cityCount {
  @include type-nav-vertical-large;

  margin-left: auto; // push right
  padding: 0 16px;
  background: $color-primary--700;

  .citySelected & {
    color: $color-primary--080;
  }
}

