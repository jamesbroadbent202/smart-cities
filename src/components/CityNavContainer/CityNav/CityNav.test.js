import React from 'react';
import { shallow } from 'enzyme';
import CityNav from './CityNav';

const defaultProps = {
  categoryId: 'mock',
  cityId: 'perth',
  cities: [
    {
      id: 'perth',
      name: 'Perth',
    },
    {
      id: 'sydney',
      name: 'Sydney',
    },
  ],
};

it('should match Snapshot', () => {
  const component = shallow(
    <CityNav {...defaultProps} />
  );

  expect(component.debug()).toMatchSnapshot();
});

it('should render the number of cities', () => {
  const component = shallow(
    <CityNav {...defaultProps} />
  );

  expect(component.find('.cityCount').text()).toBe('2');
});
