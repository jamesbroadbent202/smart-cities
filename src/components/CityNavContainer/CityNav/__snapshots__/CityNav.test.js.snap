// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
"<div className=\\"citySelected\\">
  <NavLink className=\\"link allCitiesLink\\" activeClassName=\\"link__active\\" to=\\"/all-cities/mock\\" ariaCurrent=\\"true\\">
    All Cities
    <Pill className=\\"cityCount\\">
      2
    </Pill>
  </NavLink>
  <NavLink className=\\"link cityLink\\" activeClassName=\\"link__active\\" to=\\"/perth/mock\\" ariaCurrent=\\"true\\">
    Perth
  </NavLink>
  <NavLink className=\\"link cityLink\\" activeClassName=\\"link__active\\" to=\\"/sydney/mock\\" ariaCurrent=\\"true\\">
    Sydney
  </NavLink>
</div>"
`;
